instrumentation/log4j/log4j-2.13.2/library/src/test/java/io/opentelemetry/instrumentation/log4j/v2_13_2/LogEventMapperTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.log4j.v2_13_2;
+
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.SPAN_ID;
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.TRACE_FLAGS;
+import static io.opentelemetry.instrumentation.api.log.LoggingContextConstants.TRACE_ID;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_FQCN;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_LOGGER_NAME;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_MARKER;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_NDC;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_THREAD_ID;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_THREAD_NAME;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_THREAD_PRIORITY;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_THROWABLE_MESSAGE;
+import static io.opentelemetry.instrumentation.log4j.v2_13_2.LogEventMapper.ATTR_THROWABLE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.logs.LogBuilder;
+import io.opentelemetry.sdk.logs.data.Severity;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.MarkerManager;
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.spi.DefaultThreadContextStack;
+import org.apache.logging.log4j.spi.ThreadContextStack;
+import org.apache.logging.log4j.util.ReadOnlyStringMap;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+class LogEventMapperTest {
+
+  private LogEvent logEvent;
```

Comment:
Better to use `MockitoExtension` and `@Mock` to get strict stubs

Reply:
That being said, I think we'd generally verify mapping via `LogData` instead of using mocks, i.e., these tests could be transfered into the AppenderTest. Imagine adding some helper methods or something on the builder - mocks are nice for RPC stubs, but for data classes we may as well check the actual data.

Reply:
Yeah looking at this test closer, all the code being exercised is also exercised in `OpenTelemetryAppenderConfigTest`. So I feel comfortable getting rid of this test class altogether.